pkg/api/v1/orchestrator/routes/handlers.go

```diff
@@ -0,0 +1,561 @@
+	}
+
+	conditionKind := rctypes.Kind(c.Param("conditionKind"))
+	if !r.conditionKindValid(conditionKind) {
+		r.logger.WithFields(logrus.Fields{
+			"kind": conditionKind,
+		}).Info("unsupported condition kind")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "unsupported condition kind: " + string(conditionKind),
+		}
+	}
+
+	conditionRecord, err := r.repository.Get(ctx, serverID)
+	if err != nil {
+		if errors.Is(err, store.ErrConditionNotFound) {
+			return http.StatusNotFound, &v1types.ServerResponse{
+				Message: "condition not found for server",
+			}
+		}
+
+		r.logger.WithError(err).Info("condition record query error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "condition lookup: " + err.Error(),
+		}
+	}
+
+	var activeConditionExists bool
+	for _, cond := range conditionRecord.Conditions {
+		if cond.Kind == conditionKind && !rctypes.StateIsComplete(cond.State) {
+			activeConditionExists = true
+		}
+	}
+
+	if !activeConditionExists {
+		return http.StatusNotFound, &v1types.ServerResponse{
+			Message: "no active condition found for server",
+		}
+	}
+
+	// TODO: if a task exists for this condition in an active state, don't proceed
+
+	// pop condition from the Jetstream queue
+	cond, err := r.conditionJetstream.pop(ctx, conditionKind, serverID)
+	if err != nil {
+		if errors.Is(err, errNoConditionInQueue) {
+			return http.StatusNotFound, &v1types.ServerResponse{
+				Message: "no condition in queue",
+			}
+		}
+
+		r.logger.WithError(err).Info("condition queue fetch error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "Condition Queue fetch error error: " + err.Error(),
+		}
+	}
+
+	// register this worker
+	controllerID, err := r.livenessKV.register(serverID.String())
+	if err != nil {
+		r.logger.WithField("condition.id", cond.ID).WithError(err).Info("controller registration error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "Controller registration error: " + err.Error(),
+		}
+	}
+
+	st := rctypes.NewTaskStatusRecord("controller fetched condition using client IP: " + c.RemoteIP())
+	sv := &rctypes.StatusValue{
+		WorkerID: controllerID.String(),
+		Target:   serverID.String(),
+		TraceID:  trace.SpanFromContext(ctx).SpanContext().TraceID().String(),
+		SpanID:   trace.SpanFromContext(ctx).SpanContext().SpanID().String(),
+		State:    string(rctypes.Pending),
+		Status:   st.MustMarshal(),
+	}
+
+	// create condition status value entry
+	if err := r.statusValueKV.publish(
+		r.facilityCode,
+		cond.ID,
+		controllerID,
+		cond.Kind,
+		sv,
+		true,
+		false,
+	); err != nil {
+		r.logger.WithField("condition.id", cond.ID).WithError(err).Info("status KV publish error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "Condition first status publish error: " + err.Error(),
+		}
+	}
+
+	// setup task to be published
+	task := rctypes.NewTaskFromCondition(cond)
```

Comment:
This is a design question,
The Task object includes all information that was sent in the original Condition along with Status Update records. And for Controllers like Flasher - all the work that is to be done, checkpointed - which enables Task resumption.

Long term, the Task KV will supercede and deprecate StatusValue, because it holds all required information.